test/python/synthesis/test_synthesis.py

```diff
@@ -1544,6 +1544,31 @@ def inverse(self, annotated: bool = False):
         circ = decomposer(unitary)
         self.assertEqual(Operator(unitary), Operator(circ))
 
+    @data(True, False)
+    def test_assertPickle(self, is_standard):
+        """Assert that TwoQubitControlledUDecomposer supports pickle"""
+
+        class CustomXXGate(RXXGate):
+            """Custom RXXGate subclass that's not a standard gate"""
+
+            _standard_gate = None
+
+            def __init__(self, theta, label=None):
+                super().__init__(theta, label)
+                self.name = "MyCustomXXGate"
```

Comment:
You also want to add the equality method to your custom gate class, otherwise two gates of type ``CustomXXGate`` would not be considered equal and an assert below will fail. More precisely:
```
def __eq__(self, other):
   return isinstance(other, CustomXXGate)
```

Reply:
I don't need this function and pylint asked to remove it since it's not used.